bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfile.java

```diff
@@ -139,7 +139,7 @@ public void onStateUpdateFromHandler(State state) {
         if (state instanceof StringType) {
             Optional<String> match = resolveNumber(state.toString());
             State newState = match.map(name -> (State) new StringType(name)).orElse(state);
-            if (newState == state) {
+            if (newState.equals(state)) {
```

Comment:
Can you please revert this too and add the `"PMD.CompareObjectsWithEquals"` suppression? Thanks.

Reply:
I've addressed the comment.